Assets/MixedRealityToolkit/_Core/Inspectors/Profiles/MixedRealityControllerVisualizationProfileInspector.cs

```diff
@@ -0,0 +1,278 @@
+
+                if (!hasValidType)
+                {
+                    EditorGUILayout.HelpBox("A controller type must be defined!", MessageType.Error);
+                }
+
+                var handednessValue = mixedRealityControllerHandedness.intValue - 1;
+
+                // Reset in case it was set to something other than left or right.
+                if (handednessValue < 0 || handednessValue > 1) { handednessValue = 0; }
+
+                EditorGUI.BeginChangeCheck();
+                handednessValue = EditorGUILayout.IntPopup(new GUIContent(mixedRealityControllerHandedness.displayName, mixedRealityControllerHandedness.tooltip), handednessValue, HandednessSelections, null);
+
+                if (EditorGUI.EndChangeCheck())
+                {
+                    mixedRealityControllerHandedness.intValue = handednessValue + 1;
+                }
+
+                var overrideModel = controllerSetting.FindPropertyRelative("overrideModel");
+                var overrideModelPrefab = overrideModel.objectReferenceValue as GameObject;
+
+                var controllerUseDefaultModelOverride = controllerSetting.FindPropertyRelative("useDefaultModel");
+                EditorGUILayout.PropertyField(controllerUseDefaultModelOverride);
+
+                if (controllerUseDefaultModelOverride.boolValue && overrideModelPrefab != null)
+                {
+                    EditorGUILayout.HelpBox("When default model is used, the override model will only be used if the default model cannot be loaded from the driver.", MessageType.Warning);
+                }
+
+                EditorGUI.BeginChangeCheck();
+                overrideModelPrefab = EditorGUILayout.ObjectField(new GUIContent(overrideModel.displayName, "If no override model is set, the global model is used."), overrideModelPrefab, typeof(GameObject), false) as GameObject;
+
+                if (EditorGUI.EndChangeCheck() && CheckVisualizer(overrideModelPrefab))
+                {
+                    overrideModel.objectReferenceValue = overrideModelPrefab;
+                }
+
+                EditorGUI.indentLevel--;
+            }
+        }
+
+        private bool CheckVisualizer(GameObject modelPrefab)
+        {
+            if (modelPrefab == null) { return true; }
+
+            var prefabType = PrefabUtility.GetPrefabType(modelPrefab);
+            if (prefabType != PrefabType.Prefab)
+            {
+                Debug.LogWarning("Assigned GameObject must be a prefab");
+                return false;
+            }
+
+            var componentList = modelPrefab.GetComponentsInChildren<IMixedRealityControllerVisualizer>();
+
+            if (componentList == null || componentList.Length == 0)
+            {
+                if (thisProfile.ControllerVisualizationType != null &&
+                    thisProfile.ControllerVisualizationType.Type != null)
+                {
+                    modelPrefab.AddComponent(thisProfile.ControllerVisualizationType.Type);
```

Comment:
Looks like even when the visualization script is successfully added, this still falls through to return `false` at line 275. This causes override model setting to take two steps. The first time adds the script but doesn't fill in the field.

Reply:
I'll be sure to fix. Thanks.